src/modules/api/controllers/indexer.controller.ts

```diff
@@ -1,30 +1,143 @@
-import { Post } from '@nestjs/common'
-import { Body } from '@nestjs/common'
+import {
+  BadRequestException,
+  HttpCode,
+  HttpStatus,
+  InternalServerErrorException,
+  Put,
+  Query,
+} from '@nestjs/common'
 import { Controller, Get, Param } from '@nestjs/common'
+import { ApiNotFoundResponse, ApiOkResponse, ApiQuery } from '@nestjs/swagger'
 import { indexerContainer } from '../indexer-api.module'
+import { DocumentView } from '../resources/document.view'
 
 // Need to keep a top-level container here to avoid garbage collection
 // @Controller(`${INDEXER_API_BASE_URL}/debug`)
 @Controller(`/api/v0/node/indexer`)
 export class IndexerApiController {
   constructor() {}
 
-  @Get('getdiscussion/:id')
-  async getDiscussion(@Param('id') postId) {
-    console.log(postId)
-    const out = []
-    for await (const item of indexerContainer.self.getDiscussion(postId)) {
-      out.push(item)
+  @Put('index/:documentStreamId')
+  @ApiOkResponse({ description: 'The document was found and indexed' })
+  @ApiNotFoundResponse({ description: 'The document was not found on ceramic' })
+  public async indexDocument(@Param('documentStreamId') streamId: string) {
+    throw new InternalServerErrorException('not implemented')
+  }
+
```

Comment:
BTW, indexing a document is a delay intensive/delay tolerant process. There should not be an immediate response. Rather some kind of token or way to tell if it has been indexed later.